### ISSUE TITLE
Spell check in code editor.

### ISSUE THREAD

**Issue by nanaktaev** [defold/defold#9233]
Spelling errors and typos often lead to bugs and other problems. It would be nice to have a built in spell checker in code editor.

**Comment by vlaaad** [comment]
Good idea, would be cool to implement it as a language server extension, e.g. using this one https://github.com/tekumara/typos-lsp

**Comment by vlaaad** [comment]
@nanaktaev BTW, did you know you might use some typo-related helpers today in Defold?

Defold ships Lua language server that can show warnings on type-annotated code:
<img width="313" alt="Screenshot 2024-07-30 at 11 46 29" src="https://github.com/user-attachments/assets/242c4d0c-f5d5-4ea4-bd32-67fc4edcbd39">

Here are annotation docs supported in Defold: https://luals.github.io/wiki/annotations/
